packages/office-ui-fabric-react/src/components/Callout/CalloutContent.tsx

```diff
@@ -76,7 +76,8 @@ export class CalloutContent extends BaseComponent<ICalloutProps, ICalloutState>
   }
 
   public componentWillUpdate(newProps: ICalloutProps) {
-    if (newProps.targetElement !== this.props.targetElement || newProps.target !== this.props.target) {
+    // If the target element changed, find the new one. If we are tracking target with class name, always find element because we do not know if fabric has rendered a new element and disposed the old element.
+    if (newProps.targetElement !== this.props.targetElement || newProps.targetElement == null) {
```

Comment:
@joschect Your codepen does show that the callout is not updating location when the button changes.  When the callout is opened, the beak points to the center of the button.  When it changes text and therefore width, the beak is not moving with it.  In Planner this manifests by the callout moving to the upper left corner of the screen.